llama_index/vector_stores/epsilla.py

```diff
@@ -0,0 +1,271 @@
+            self._collection_created = True
+
+        if self._collection_name in table_list and overwrite is True:
+            status_code, response = self._client.drop_table(
+                table_name=self._collection_name
+            )
+            if status_code != 200:
+                self._handle_error(msg=response["message"])
+            logger.debug(
+                f"Successfully removed old collection: {self._collection_name}"
+            )
+            if dimension is not None:
+                self._create_collection(dimension)
+
+        if self._collection_name not in table_list and dimension is not None:
+            self._create_collection(dimension)
+
+    def client(self) -> Any:
+        """Return the Epsilla client."""
+        return self._client
+
+    def _handle_error(self, msg: str) -> None:
+        """Handle error."""
+        logger.error(f"Failed to get records: {msg}")
+        raise Exception("Error: {}.".format(msg))
+
+    def _create_collection(self, dimension: int) -> None:
+        """
+        Create collection.
+
+        Args:
+            dimension (int): The dimension of the embeddings.
+        """
+        fields: List[dict] = [
+            {"name": "id", "dataType": "STRING"},
+            {"name": DEFAULT_DOC_ID_KEY, "dataType": "STRING", "primaryKey": True},
+            {"name": DEFAULT_TEXT_KEY, "dataType": "STRING"},
+            {
+                "name": DEFAULT_EMBEDDING_KEY,
+                "dataType": "VECTOR_FLOAT",
+                "dimensions": dimension,
+            },
+            {"name": "metadata", "dataType": "JSON"},
+        ]
+        status_code, response = self._client.create_table(
+            table_name=self._collection_name, table_fields=fields
+        )
+        if status_code != 200:
+            self._handle_error(msg=response["message"])
+        self._collection_created = True
+        logger.debug(f"Successfully created collection: {self._collection_name}")
+
+    def add(
+        self,
+        embedding_results: List[NodeWithEmbedding],
+    ) -> List[str]:
+        """
+        Add embedding results to Epsilla vector store.
+
+        Args
+            embedding_results: List[NodeWithEmbedding]: list of embedding results.
+
+        Returns:
+            List[str]: List of ids inserted.
+        """
+        # If the collection doesnt exist yet, create the collection
+        if not self._collection_created and len(embedding_results) > 0:
+            dimension = len(embedding_results[0].embedding)
+            self._create_collection(dimension)
+
+        elif len(embedding_results) == 0:
+            return []
+
+        ids = []
+        records = []
+        for result in embedding_results:
+            ids.append(result.id)
+            text = result.node.get_content(metadata_mode=MetadataMode.NONE)
+            metadata_dict = node_to_metadata_dict(result.node, remove_text=True)
+            metadata = metadata_dict["_node_content"]
```

Comment:
curious why this step? The `_node_content` is not the actual metadata, it's a serialized version of the node so that we can re-construct it when retrieved

The `metadata_dict` contains actual metadata as well that is removed here

Reply:
@logan-markewich When I checked the `node_to_metadata_dict` function, the `metadata["_node_content"]` seems a json string dumping from the node_dict, excluding text, id, doc_id and etc, which is exactly what we need to insert for metadata(if you check the schema we have). Please let me know if I understand it wrongly. Thanks.

Reply:
If the filtering works for you, then it works! Was just curious :)